akka-tools-persistence/src/main/scala/no/nextgentel/oss/akkatools/persistence/EnhancedPersistentActor.scala

```diff
@@ -37,24 +37,73 @@ abstract class EnhancedPersistentActor[E:ClassTag, Ex <: Exception : ClassTag]
 
   implicit val ec = context.dispatcher
 
-  private   var isProcessingEvent = false
-  private   var pendingDurableMessage:Option[DurableMessage] = None
-  private   var timeoutTimer:Option[Cancellable] = None
+  case class InternalState(
+                            // Used when processing events live - not recovering
+                            var eventLogLevelInfo: Boolean,
+
+                            // Used when recovering events
+                            var recoveringEventLogLevelInfo: Boolean,
+
+                            // Used when processing commands
+                            var cmdLogLevelInfo: Boolean,

```

Comment:
The three ...LogLevelInfo can be moved out of "state".. Set runtime and only affects logging (slf4j)j